### PR TITLE
SAMZA-1161: Adding metrics into LocalStoreMonitor.

### PR DESCRIPTION
**Pull request by shanthoosh**
Rocksdb LocalStoreMonitor is responsible for clearing up unused local task partition stores. Metrics are required to understand the behavior of this monitor in production(especially when it's clearing up unused rocksdb local stores).

The following two metrics will be emitted from this monitor:
a) Total disk space cleared in bytes.
b) Total number of rocksdb task partition stores cleared.

### CODE REVIEW

**Comment by shanthoosh** [comment]
@jmakes : Please take a look when you get a chance.

File: samza-rest/src/main/java/org/apache/samza/monitor/LocalStoreMonitorConfig.java [hunk 1]
```diff
@@ -46,7 +46,7 @@
    * Defines the comma separated list of job status servers of the form
    * "Host1:Port1,Host2:Port2".
    */
-  private static final String CONFIG_JOB_STATUS_SERVERS = "job.status.servers";
+  private static final String CONFIG_JOB_STATUS_SERVERS = "jobs.status.servers";
```
Comment (jmakes) [inline samza-rest/src/main/java/org/apache/samza/monitor/LocalStoreMonitorConfig.java]: This makes it inconsistent with the properties in JobsResourceConfig and therefore more confusing for users. Why is this change desirable?

File: samza-rest/src/main/java/org/apache/samza/monitor/LocalStoreMonitor.java [hunk 2]
```diff
@@ -147,6 +145,8 @@ private void markSweepTaskStore(File taskStoreDir) throws IOException {
     if (!offsetFile.exists()) {
       LOG.info("Deleting the task store : {}, since it has no offset file.", taskStorePath);
       FileUtils.deleteDirectory(taskStoreDir);
+      localStoreMonitorMetrics.diskSpaceFreedInBytes.inc(taskStoreDir.getTotalSpace());
```
Comment (jmakes) [inline samza-rest/src/main/java/org/apache/samza/monitor/LocalStoreMonitor.java]: Does getTotalSpace() return an appropriate value AFTER the dir has already been deleted (line 147)?

How was this tested?

File: samza-rest/src/test/java/org/apache/samza/monitor/TestLocalStoreMonitor.java [hunk 3]
```diff
@@ -65,7 +65,7 @@ public void setUp() throws Exception {
 
     // Initialize the local store monitor with mock and config
     localStoreMonitor = new LocalStoreMonitor(new LocalStoreMonitorConfig(new MapConfig(config)),
-                                              new NoOpMetricsRegistry(),
+                                              new LocalStoreMonitorMetrics("TestMonitorName", new NoOpMetricsRegistry()),
```
Comment (jmakes) [inline samza-rest/src/test/java/org/apache/samza/monitor/TestLocalStoreMonitor.java]: Maybe NoOpMetricsRegistry is no longer appropriate and the new metrics should be tested.

File: samza-rest/src/main/java/org/apache/samza/monitor/LocalStoreMonitor.java (hunk 2 above)
Comment (shanthoosh) [inline samza-rest/src/main/java/org/apache/samza/monitor/LocalStoreMonitor.java]: This was tested in my devbox(execution environment is on open-jdk). 

`Does getTotalSpace() return appropriate value?`

It depends upon the JDK/JVM implementation of File interface. In some implementations, the size of the File handle in bytes is read(along with other file attributes) when the contents are read/written(which is inline with the documentation for this interface). My sample test totally separate from this work fine with that.  

In some implementations, this total bytes value is fetched during each call of this API(no pre-fetch & storage during read/write call). 

 I'm moving this getTotalSpace() call before the delete operation.

File: samza-rest/src/test/java/org/apache/samza/monitor/TestLocalStoreMonitor.java (hunk 3 above)
Comment (shanthoosh) [inline samza-rest/src/test/java/org/apache/samza/monitor/TestLocalStoreMonitor.java]: Added more assertions based upon the metrics generated.

File: samza-rest/src/main/java/org/apache/samza/monitor/LocalStoreMonitorConfig.java (hunk 1 above)
Comment (shanthoosh) [inline samza-rest/src/main/java/org/apache/samza/monitor/LocalStoreMonitorConfig.java]: The recent for renaming is for consistency. 

All other config keys that are in this Config class have prefix jobs(and this has prefix job). 
* How this collides with JobsResourceConfig? 
* Why this is confusing? (Reason for extracting this out separately is to group all LocalStore Monitor config together. May be we could consider renaming all the config within this class to keep it different from JobsResourceConfig).

Comment (jmakes) [inline samza-rest/src/main/java/org/apache/samza/monitor/LocalStoreMonitorConfig.java]: Ahh didn't see those other properties in LocalStoreMonitorConfig, so from that perspective it makes sense. But... 

> How this collides with JobsResourceConfig?
> Why this is confusing?
 
Not collide, inconsistent. If a user configures samza-rest with both the jobs resource and this monitor they will have a mix of properties using "job." and "jobs." to refer to the same thing, right? 
`
job.proxy.factory.class=org.apache.samza.myClass
job.installations.path=./
jobs.status.servers=128.0.0.1
`
That will seem odd, and users will be unsure if there's some subtle difference they're missing.

Comment (shanthoosh) [inline samza-rest/src/main/java/org/apache/samza/monitor/LocalStoreMonitorConfig.java]: Yes, that wouldn't make less sense to the user (when using both the resources together). 

Is it reasonable to change the prefix of all the config keys in `LocalStoreMonitorConfig` from 'jobs' to 'job' (we'll have consistency across all the config keys in `samza-rest`)

Comment (jmakes) [inline samza-rest/src/main/java/org/apache/samza/monitor/LocalStoreMonitorConfig.java]: That's probably the best option. LocalStoreMonitor is newer than JobsResource, so it has had less exposure and will have less/no impact on users.

Comment (shanthoosh) [inline samza-rest/src/main/java/org/apache/samza/monitor/LocalStoreMonitorConfig.java]: Incorporated this change in the upcoming commit.

**Comment by shanthoosh** [comment]
@jmakes 
Please take a look when you get a chance.

The build was successful (https://builds.apache.org/job/samza-pr-checks/111/), however the github-jenkins plugin doesn't update the result here.

**Pull request closed without merging**